src/test/java/com/exactpro/junit5/TestFormatter.java

```diff
@@ -3,63 +3,46 @@
 import io.netty.buffer.ByteBuf;
 import org.junit.jupiter.api.Test;
 
-import java.io.IOException;
 import java.util.ArrayList;
 import java.util.List;
 
 import static org.junit.jupiter.api.Assertions.*;
 
 class TestFormatter {
 
+    private String methodName = "aMethod";
+
     @Test
-    void testNameOneByte() throws IOException {
+    void testNameOneByte() {
         String hex = "12";
-        String methodName = new Object() {
-        }
-            .getClass()
-            .getEnclosingMethod()
-            .getName();
-
         List<ByteBuf> arguments = new ArrayList<>();
         arguments.add(ByteBufUtils.fromHex(hex));
         WithByteBufInvocationContext invocationContext =
             new WithByteBufInvocationContext(methodName, new WithByteBufMethodContext(), arguments);
-        assertEquals(methodName + " [ " + hex + " ]", invocationContext.getDisplayName(1));
+        assertEquals("aMethod" + " [ " + hex + " ]", invocationContext.getDisplayName(1));
     }
 
     @Test
-    void testNameMultipleByteOneBuffer() throws IOException {
-        String hex = "12 13 a4";
-        String methodName = new Object() {
-        }
-            .getClass()
-            .getEnclosingMethod()
-            .getName();
-
+    void testNameMultipleByteOneBuffer() {
+        String hex = "12 13 A4";
         List<ByteBuf> arguments = new ArrayList<>();
         arguments.add(ByteBufUtils.fromHex(hex));
         WithByteBufInvocationContext invocationContext =
             new WithByteBufInvocationContext(methodName, new WithByteBufMethodContext(), arguments);
-        assertEquals(methodName + " [ " + hex + " ]", invocationContext.getDisplayName(1));
+        assertEquals("aMethod" + " [ " + hex + " ]", invocationContext.getDisplayName(1));
     }
 
     @Test
-    void testNameMultipleByteMultipleBuffer() throws IOException {
+    void testNameMultipleByteMultipleBuffer() {
         String[] hexes = {"12 13", "a4", "11"};
-        String methodName = new Object() {
-        }
-            .getClass()
-            .getEnclosingMethod()
-            .getName();
-
         List<ByteBuf> arguments = new ArrayList<>();
         for (String hex : hexes) {
```

Comment:
move this code to separate static method. reuse it in all tests.

Reply:
can I make it as a private method in the same class ?

Reply:
yes, It's exactly that I've asked